stairsplus/common.lua

```diff
@@ -8,15 +8,15 @@ Licensed under the zlib license. See LICENSE.md for more information.
 local S = moreblocks.S
 
 local descriptions = {
-	["micro"] = S("%s Microblock"),
-	["slab"] = S("%s Slab"),
-	["slope"] = S("%s Slope"),
-	["panel"] = S("%s Panel"),
-	["stair"] = S("%s Stairs"),
+	["micro"] = "Microblock",
+	["slab"] = "Slab",
+	["slope"] = "Slope",
+	["panel"] = "Panel",
+	["stair"] = "Stairs",
 }
 
 stairsplus.register_single = function(category, alternate, info, modname, subname, recipeitem, fields)
-	local desc_base = descriptions[category]:format(fields.description)
+	local desc_base = S("@1 "..descriptions[category], fields.description)
```

Comment:
It is better avoiding concatenation.

Reply:
What would be the recommended way? I have removed it to avoid the error: ignoring escape sequence 'Tmoreblocks' in translation

Reply:
In this kind of function, it is better when _full_ translated description can be given as argument as in some languages `descriptions[category]` and `fields.description` are not always in the same order as in english (and also `fields.description` value can depend on value of `descriptions[category]` in languages like french) `.
It is ok to keep old parameters and adding an optional new one, and if if the full translated description is nil, fallback to not translated string else description is set to the new parameter. This implies refactoring a bit this function.

Reply:
Is this what you mean?
S("@1 @2", fields.description, S(descriptions[category]))
I would have to change
@1 Microblock=Microbloque de @1
to
Microblock=Microbloque
@1 @2=@2 de @1

Reply:
No, `@1 @2` doesn't allow to have values of `@1` and `@2` depending on each other.
I mean, something like
```lua
 stairsplus.register_single = function(category, alternate, info, modname, subname, recipeitem, fields, fulldescription)
```
and
```lua
local desc_base = fields.description..descriptions[category]
-- …
-- if (…) then
-- …
def.description = fulldescription or ("%s (%d/16)"):format(desc_base, info)
-- …
-- else
-- …
def.description = fulldescription and S("@1 (@2/16)", fulldescription, compute_at_2()) or desc_base .. alternate:gsub("_", " "):gsub("(%a)(%S*)", function(a, b) return a:upper() .. b end)
```
and each `something Microblock` should be fully added in tr files (yes, it is a lot (they are all in stairsplus/registrations.lua), but have a look at translation files of `minetest_game/stairs` for reference). I can create the resulting template.txt if you want + french translation (but I can not have perfect translation in other languages because word order may varies and words may have different grammatical form depending on categories)
This means stairplus api must be changed, and all its calls… I can also help with this if you need.

Reply:
The example above if it works, can't translate all the elements because it depends on the mods installed, on my server there would be 9112 blocks in moreblocks
![Screenshot_20200313_151923](https://user-images.githubusercontent.com/61398224/76649424-6a5c4b80-653f-11ea-9ea0-7fa94c93089c.png)

Reply:
Can you give me some examples of mods defining nodes in moreblocks, please (or the name of your server if it is a public one)? In the next couple of days, I will try to implement a backward compatible solution allowing to have better translations if external mods define specific translation strings, but defaulting to same as your implementation. I hope, this would allow all your mods to work well in your language.

Reply:
mods: 
- bridger
- scifi_nodes
- concrete (the one in the image above)
- building_blocks

Reply:
Is 86502f295ac5e622851a0d5afb9b82b307989907 working well with your mods? I have tried with technic and can reproduce your screenshot. Some strings like `@1 Stair Outer` still need to be updated in .tr files.

Reply:
It works fine, there are missing strings to translate, but I still don't understand what fulldescs does, I don't notice the difference